src/Components/LeftPanel/DataSourceList.jsx

```diff
@@ -61,11 +69,92 @@ export const DataSourceList = ({ isLoading, dataSource, selectedFaction, setSele
       mobile: { ...settings.mobile, closedRoles: newClosedRoles },
     });
   };
+  const handleAddCardToCategoryClick = (card, category = undefined) => {
+    switch (card.type) {
+      case "role":
+        // add cards that belong to this role
+        dataSource.filter((c) => c.role === card.name).forEach((i) => onAddToCategory(category, i));
+        break;
+      case undefined:
+        onAddToCategory(category, card);
+        break;
+      default:
+      case "header":
+      case "category":
+      case "allied":
+        break;
+    }
+  };
+  const [contextMenu, setContextMenu] = useState(null);
+
+  const handleContextMenu = (e, card) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (card.type === undefined) {
+      setContextMenu({
+        x: e.clientX,
+        y: e.clientY,
+        items: [
+          {
+            key: "clicked-item",
+            label: <b>{card.name}</b>,
+            disabled: true,
+          },
+          {
+            key: "add-single",
+            label: (
+              <Dropdown
+                getPopupContainer={(node) => node}
+                menu={{
+                  items: buildCategoryMenuItems(categories),
+                  onClick: (e) => handleAddCardToCategoryClick(card, e.key),
+                }}>
+                <div>Add this item to</div>
+              </Dropdown>
+            ),
+            icon: <CirclePlus size={14} />,
+          },
+        ],
+      });
+    } else if (card.type === "role") {
+      setContextMenu({
+        x: e.clientX,
+        y: e.clientY,
+        items: [
+          {
+            key: "clicked-item",
+            label: <b>{card.name}</b>,
+            disabled: true,
+          },
+          {
+            key: "add-all",
+            label: (
+              <Dropdown
+                getPopupContainer={(node) => node}
+                menu={{
+                  items: buildCategoryMenuItems(categories),
+                  onClick: (e) => handleAddCardToCategoryClick(card, e.key),
+                }}>
+                <div>Add all items of this category to</div>
```

Comment:
```suggestion
                <div>Add all items to..</div>
```
_Less is more_ is probably a think here.